### PR TITLE
Feature/cold storage files implementation

### PR DESCRIPTION
**Pull request by udaykumar1995**
Implemented cold storage files implementation

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -323,6 +337,10 @@ def upload_csv():
     data["transfer_source"] = ""
     data["transfer_destination"] = ""
 
+    data["sealed_state"] = data["sealed_state"].map(
```
Comment (NISH1001) [inline src/app.py:340]: Does `sealed_state` exist in the incoming csv? If it doesn't exists, how will this be handled?

File: src/constants.py [hunk 2]
```diff
@@ -15,6 +17,16 @@
     "ContentDate:End": "content_date_end",
     "Checksum:Algorithm": "checksum_algorithm",
     "Checksum:Value": "checksum_value",
+    "IsSealed": "sealed_state",
```
Comment (NISH1001) [inline src/constants.py:20]: This is for POST requests, creating a single catalog item in the table. Is it required?  @ub0005 Can you please double-check with @DImuthuUpe

File: src/services/db/models.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ class CatalogueItem(db.Model):
     checksum_algorithm = db.Column(db.String)
     checksum_value = db.Column(db.String)
 
+    transfer_id = db.Column(db.String)
```
Comment (NISH1001) [inline src/services/db/models.py:34]: Is `transfer_id` required? (I might be out of context). Isn't file uuid feasible to do the tracking?

File: src/constants.py (hunk 2 above)
Comment (udaykumar1995) [reply src/constants.py:20]: we are using CATALOGUE_CSV_COLUMN_MAPPER in bulk upload @NISH1001 
Do you want to check whether IsSealed is needed as a mandatory value to send?

File: src/app.py (hunk 1 above)
Comment (udaykumar1995) [reply src/app.py:340]: yes, isSealed should be part of uploading CSV. I will check with Dimuthu if it is a mandatory column to send in CSV. 
If it's mandatory then I will include a check to ensure it's not empty.

File: src/services/db/models.py (hunk 3 above)
Comment (NISH1001) [reply src/services/db/models.py:34]: Edit: Got it....

File: src/app.py (hunk 1 above)
Comment (NISH1001) [reply src/app.py:340]: That makes sense. Let's do a sanity check in CSV itself. If the column doesn't exist, we can add a default one?

Comment (ub0005) [reply src/app.py:376]: Making the IsSealed column a mandatory column and value in the CSV. 
Added the checks in the code as well. 
@NISH1001

Comment (NISH1001) [reply src/app.py:376]: @ub0005 Eh thanks... I saw the code. The checks should work...

File: src/constants.py [hunk 4]
```diff
@@ -15,6 +17,16 @@
...
+}
+
+CATALOGUE_SEALED_STATE_MAPPER = {
```
Comment (NISH1001) [inline src/constants.py:23]: @udaykumar1995 Is there a better way to handle the mapping?

**Comment by NISH1001** [comment]
@ub0005 Overall the PR looks good. @xhagrg

File: src/constants.py (hunk 4 above)
Comment (ub0005) [reply src/constants.py:23]: I am not sure, do you see any drawbacks with this approach? @NISH1001

Comment (NISH1001) [reply src/constants.py:23]: Eh no. I don't see any drawbacks. But I was thinking if dictionary-based is better or function-based approach. For now, let's just go with the current one with dictionary...

File: src/app.py [hunk 5]
```diff
@@ -291,11 +338,13 @@ def upload_csv():
         )
 
     # make sure these columns aren't empty
-    if data[["uuid", "name"]].isna().sum().sum() > 0:
-        logger.error("uuid or name column values are empty!")
+    if data[["uuid", "name", "sealed_state"]].isna().sum().sum() > 0:
```
Comment (xhagrg) [inline src/app.py:341]: Why double sum?

**Comment by xhagrg** [comment]
Just a small comment, other than that it looks good.

File: src/app.py (hunk 5 above)
Comment (NISH1001) [reply src/app.py:341]: This is to make sure any of the given columns are not null. the first sum() will give us the sum per column. And applying sum again to that gives us a single value.

**Pull request merged**